airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/writer.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from airbyte_cdk.models import AirbyteStream
+from pygsheets import Worksheet
+
+from .buffer import WriteBuffer
+from .spreadsheet import GoogleSheets
+
+
+class GoogleSheetsWriter(WriteBuffer):
+    def __init__(self, spreadsheet: GoogleSheets):
+        self.spreadsheet = spreadsheet
+
+    def delete_stream_entries(self, stream_name: str):
+        """
+        Deletes all the records belonging to the input stream.
+        """
+        self.spreadsheet.clean_worksheet(f"{stream_name}")
+
+    def check_headers(self, stream_name: str):
+        """
+        Checks whether data headers belonging to the input stream are set.
+        """
+        for streams in self.stream_info:
+            if stream_name in streams:
+                if not streams["is_set"]:
+                    self.spreadsheet.set_headers(stream_name, streams[stream_name])
+                    streams["is_set"] = True
+
+    def queue_write_operation(self, stream_name: str):
+        """
+        Mimics `batch_write` operation using records_buffer.
+
+        1) gets data from the records_buffer
+        2) writes it to the target worksheet
+        3) cleans-up the records_buffer belonging to input stream
+        """
+        for streams in self.records_buffer:
+            if stream_name in streams:
+                if len(streams[stream_name]) == self.flush_interval:
+                    self.write_from_queue(stream_name)
+                    self.flush_buffer(stream_name)
+
+    def write_from_queue(self, stream_name: str):
+        """
+        Writes data from records_buffer for belonging to the input stream.
+
+        1) checks the headers are set
+        2) gets the values from the records_buffer
+        3) if there are records to write - writes them to the target worksheet
+        """
+        values: list = []
+        self.check_headers(stream_name)
+
+        for streams in self.records_buffer:
+            if stream_name in streams:
+                values = streams[stream_name]
+
+        if len(values) > 0:
+            stream: Worksheet = self.spreadsheet.open_worksheet(f"{stream_name}")
+            self.logger.info(f"Writing data for stream: {stream_name}")
+            stream.append_table(values, start="A2", dimension="ROWS")
+        else:
+            self.logger.info(f"Skipping empty stream: {stream_name}")
+
+    def write_whats_left(self):
+        """
+        Stands for writing records that are still left to be written,
+        but don't match the condition for `queue_write_operation`.
+        """
+        for streams in self.records_buffer:
+            stream_name = list(streams.keys())[0]
```

Comment:
we do we just take first stream name? can there be more or zero ?

Reply:
Reworked. Done.